server/src/test/java/org/opensearch/search/aggregations/startree/DateRangeQueryTests.java

```diff
@@ -0,0 +1,274 @@
+            Document doc = new Document();
+            if (random.nextBoolean()) {
+                val = random.nextInt(10); // Random int between 0 and 9 for status
+                doc.add(new SortedNumericDocValuesField(STATUS, val));
+            }
+            if (random.nextBoolean()) {
+                val = random.nextInt(100); // Random int between 0 and 99 for size
+                doc.add(new SortedNumericDocValuesField(SIZE, val));
+            }
+            date = nowMillis - random.nextLong(duration180DaysMillis); // Random date within last 180 days
+            doc.add(new SortedNumericDocValuesField(TIMESTAMP_FIELD, date));
+            doc.add(new LongPoint(TIMESTAMP_FIELD, date));
+            iw.addDocument(doc);
+            docs.add(doc);
+        }
+
+        if (randomBoolean()) {
+            iw.forceMerge(1);
+        }
+        iw.close();
+
+        DirectoryReader ir = DirectoryReader.open(directory);
+        initValuesSourceRegistry();
+        LeafReaderContext context = ir.leaves().get(0);
+
+        SegmentReader reader = Lucene.segmentReader(context.reader());
+        IndexSearcher indexSearcher = newSearcher(reader, false, false);
+        CompositeIndexReader starTreeDocValuesReader = (CompositeIndexReader) reader.getDocValuesReader();
+
+        List<CompositeIndexFieldInfo> compositeIndexFields = starTreeDocValuesReader.getCompositeIndexFields();
+        CompositeIndexFieldInfo starTree = compositeIndexFields.get(0);
+
+        LinkedHashMap<Dimension, MappedFieldType> supportedDimensions = new LinkedHashMap<>();
+        supportedDimensions.put(
+            new NumericDimension(STATUS),
+            new NumberFieldMapper.NumberFieldType(STATUS, NumberFieldMapper.NumberType.INTEGER)
+        );
+        supportedDimensions.put(
+            new NumericDimension(SIZE),
+            new NumberFieldMapper.NumberFieldType(SIZE, NumberFieldMapper.NumberType.INTEGER)
+        );
+        supportedDimensions.put(
+            new DateDimension(
+                TIMESTAMP_FIELD,
+                List.of(
+                    new DateTimeUnitAdapter(Rounding.DateTimeUnit.MONTH_OF_YEAR),
+                    new DateTimeUnitAdapter(Rounding.DateTimeUnit.DAY_OF_MONTH)
+                ),
+                DateFieldMapper.Resolution.MILLISECONDS
+            ),
+            new DateFieldMapper.DateFieldType(TIMESTAMP_FIELD)
+        );
+
+        DateTimeFormatter formatter = DateTimeFormatter.ISO_INSTANT;
+        Instant startOfTodayUTC = now.truncatedTo(ChronoUnit.DAYS);
+
+        long to = startOfTodayUTC.toEpochMilli();
+        long from = to - TimeUnit.DAYS.toMillis(80);
+        String fromDateString = formatter.format(Instant.ofEpochMilli(from));
+        String toDateString = formatter.format(Instant.ofEpochMilli(to));
+
+        String fromDateStringNow = "now-80d/d";
+        String toDateStringNow = "now/d";
+
+        Query query = new IndexOrDocValuesQuery(
```

Comment:
The query `from` and `to` should be randomized as well ? 

Then it would also cover the term/terms query case.

Reply:
The issue with randomization is that we do not validate at aggregator level whether a query can be resolved via star-tree or not - we do this validation while creating StarTreeFilters.
So basically with random values, we cannot ensure valid star-tree supported query types. So sending out valid and DETERMINISTIC queries here in test.

I can re-evaluate based on the test coverage if we need additional cases for 100% coverage of code logic.

Reply:
> So basically with random values, we cannot ensure valid star-tree supported query types.

Can we randomize and keep it a valid query ?

Reply:
I was able to do some randomization keeping it a valid query.